tests/cases/push-generator.js

```diff
@@ -674,15 +677,28 @@ PushGenerator.prototype.displayMessage = function() {
   this.createRequest().then(function(request) {
     var content = document.getElementById('message-info-dialog').cloneNode(true /* deep */);
     var headers = [];
+    var curlCommand = 'curl "' + request.url + '" --request POST';
 
     Object.keys(request.headers).forEach(function(headerName) {
       headers.push(headerName + ': ' + request.headers[headerName]);
+      curlCommand += ' --header "' + headerName + ': ' + request.headers[headerName] + '"';
     });
 
     content.querySelector('#endpoint').textContent = request.url;
     content.querySelector('#headers').textContent = headers.join('\n');
     content.querySelector('#body').textContent = request.body;
 
+    if (request.body instanceof ArrayBuffer) {
+      curlError = 'Since web push protocol requires a stream as ' +
+          'the body of the request, there is no CURL command that will ' +
+          'stream an encrypted payload.';
+      content.querySelector('#error').textContent = curlError;
+    } else {
+      // content-length is needed for the curl request.
+      curlCommand += ' --header "Content-Length: 0"';
+      content.querySelector('#request').textContent = curlCommand;
```

Comment:
I don't think there's an element having ID `request`. Is there a HTML part to this change?